### ISSUE TITLE
Build error: No entry for: "async"

### ISSUE THREAD

**Issue by koslambrou** [neovimhaskell/nvim-hs#47]
There is a closed issue that had almost the same error as myself. (#36)
```
Building nvim-hs-0.0.7...
Preprocessing library nvim-hs-0.0.7...

library/Neovim/Classes.hs:4:14: Warning: 
    -XOverlappingInstances is deprecated: instead use per-instance pragmas OVERLAPPING/OVERLAPPABLE/OVERLAPS
[ 1 of 27] Compiling Neovim.Log       ( library/Neovim/Log.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/Log.o )
[ 2 of 27] Compiling Neovim.Plugin.Startup ( library/Neovim/Plugin/Startup.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/Plugin/Startup.o )
[ 3 of 27] Compiling Neovim.Classes   ( library/Neovim/Classes.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/Classes.o )
[ 4 of 27] Compiling Neovim.Plugin.Classes ( library/Neovim/Plugin/Classes.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/Plugin/Classes.o )
[ 5 of 27] Compiling Neovim.Plugin.IPC.Classes ( library/Neovim/Plugin/IPC/Classes.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/Plugin/IPC/Classes.o )

library/Neovim/Plugin/IPC/Classes.hs:33:1: Warning:
    The import of ‘Data.Time.Locale.Compat’ is redundant
      except perhaps to import instances from ‘Data.Time.Locale.Compat’
    To import instances alone, use: import Data.Time.Locale.Compat()
[ 6 of 27] Compiling Neovim.Plugin.IPC ( library/Neovim/Plugin/IPC.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/Plugin/IPC.o )
[ 7 of 27] Compiling Neovim.Context.Internal ( library/Neovim/Context/Internal.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/Context/Internal.o )
[ 8 of 27] Compiling Neovim.Context   ( library/Neovim/Context.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/Context.o )
[ 9 of 27] Compiling Neovim.Plugin.Internal ( library/Neovim/Plugin/Internal.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/Plugin/Internal.o )
[10 of 27] Compiling Neovim.Config    ( library/Neovim/Config.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/Config.o )
[11 of 27] Compiling Neovim.Util      ( library/Neovim/Util.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/Util.o )
[12 of 27] Compiling Neovim.RPC.Common ( library/Neovim/RPC/Common.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/RPC/Common.o )
[13 of 27] Compiling Neovim.RPC.Classes ( library/Neovim/RPC/Classes.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/RPC/Classes.o )
[14 of 27] Compiling Neovim.RPC.FunctionCall ( library/Neovim/RPC/FunctionCall.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/RPC/FunctionCall.o )
[15 of 27] Compiling Neovim.RPC.EventHandler ( library/Neovim/RPC/EventHandler.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/RPC/EventHandler.o )
[16 of 27] Compiling Neovim.API.Parser ( library/Neovim/API/Parser.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/API/Parser.o )
[17 of 27] Compiling Neovim.API.TH    ( library/Neovim/API/TH.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/API/TH.o )
[18 of 27] Compiling Neovim.API.String ( library/Neovim/API/String.hs, dist/dist-sandbox-57d0b3ff/build/Neovim/API/String.o )

library/Neovim/API/String.hs:23:3: No entry for: "async"
cabal: Error: some packages failed to install:
nvim-hs-0.0.7 failed during the building phase. The exception was:
ExitFailure 1
```
The solution was to install a more recent version of neovim.

However, I installed the latest version as described in [Neovim install instructions](https://github.com/neovim/neovim/wiki/Installing-Neovim) which is 0.2.0-dev and got this error.

What is wrong?

**Comment by saep** [comment]
Thanks for reporting this!

I almost haven't followed the development of neovim in the past year, but it seems that the exposed remote API has changed its format and fields slightly.

I skimmed through some other API generators and it seems that the "async" entry isn't mandatory anymore and defaults to `false`. The same goes for exceptions which now seem to be possible for remote functions by default.

I just pushed the branch `lts-7.9` which should build (at least with stack), but I will spent some time this weekend to make a few working(tm) releases for all or some of the neovim beta release versions before merging that to master.
